src/command.h

```diff
@@ -91,6 +91,36 @@ class CommandBinding {
     {
     }
 
+    /** Binding to a combined command invokation and completion
+     *  function.
+     */
+    CommandBinding(std::function<void(CallOrComplete)> callOrCompl)
+    {
+        command = [callOrCompl](Input input, Output output) -> int {
+            std::pair<Input, Output> io = {input, output};
+            int exitCode = 0;
+            CallOrComplete invoc;
+            invoc.command_ = input.command();
+            invoc.inputOutput_ = &io;
+            invoc.exitCode_ = &exitCode;
+            callOrCompl(invoc);
+            return exitCode;
+        };
+        completion_ = [callOrCompl](Completion& complete) {
+            CallOrComplete invoc;
+            invoc.complete_ = &complete;
+            callOrCompl(invoc);
+        };
+    }
+
+    template <typename ClassName>
+    CommandBinding(ClassName* object,
+                   void(ClassName::*member)(CallOrComplete))
+        : CommandBinding(std::bind(member, object,
+                            std::placeholders::_1))
+    {
+    }
+
     // FIXME: Remove after C++ transition
```

Comment:
Not related to this PR, but Is this possible yet?

Reply:
There are still a few in main.cpp, but I hope I can migrate them soon.